ouroboros-consensus/src-win32/Ouroboros/Storage/IO.hs

```diff
@@ -74,6 +75,18 @@ read fh bytes = withOpenHandle "read" fh $ \h ->
   Internal.createUptoN (fromIntegral bytes) $ \ptr ->
     fromIntegral <$> win32_ReadFile h ptr (fromIntegral bytes) Nothing
 
+getCurrentFileOffset :: HANDLE -> IO Int64
+getCurrentFileOffset h = setFilePointerEx h 0 fILE_CURRENT
+
+pread :: FHandle -> Word64 -> Int64 -> IO ByteString
+pread fh bytes pos = withOpenHandle "pread" fh $ \h ->
+  Internal.createUptoN (fromIntegral bytes) $ \ptr -> do
+    initialOffset <- getCurrentFileOffset h
+    _ <- setFilePointerEx h pos fILE_BEGIN
+    n <- fromIntegral <$> win32_ReadFile h ptr (fromIntegral bytes) Nothing
+    _ <- setFilePointerEx h initialOffset fILE_BEGIN
```

Comment:
Hmm, this is unfortunate.

Reply:
Yes, unfortunately `ReadFile` always moves the file offset based on my tests and [this](https://stackoverflow.com/a/14448717)

Reply:
Note to self: currently, handles are accessed using `withMVar`, so access is serialised/thread-safe already. In the future, we will try to relax this to allow multiple `pread`s via the same handle, but only on Unix.